### ISSUE TITLE
[UI] Cluster Templates - Create cluster with template, edit and switch to another revision, PUT has old clusterTemplateId

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#21680]
Version: master-head (7/22/19)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Create a cluster template, do the following:
 Allow override for k8s version, and set value to v1.13.x
- Leverage the template to create a cluster (without changing the k8s version). Save to create cluster.
- Clone the cluster template, in order to make a second revision, do the following:
 DISABLE override for k8s version, and change the value to v1.14.4-rancher1-1 and save changes to the cluster template. (You may run into rancher/rancher#21679 issue but save anyway; it will work even if UI claims it didn't).
- Last, edit the cluster, change the cluster template to the second revision. Save.

**Result:**
The cluster won't update. It should have updated and changed k8s version to v1.14.4 but it doesn't appear to do anything.
**PUT appears to be sending the OLD revision (our first revision)**
Rancher logs show an error immediately when save:
`E0722 19:18:01.163507       6 status.go:64] apiserver received an error that is not an metav1.Status: &runtime.missingKindErr{data:"object has no kind field "}                                                                                                              
2019/07/22 19:18:01 [ERROR] ClusterController c-wnzqs [cluster-provisioner-controller] failed with : Failed to update cluster []: Operation cannot be fulfilled on clusters.management.cattle.io "c-wnzqs": the object has been modified; please apply your changes to the latest version and try again`

**Other details that may be helpful:**
single-node rancher, master-head (7/22/19) (`d8bf0e2f`)

**Comment by prachidamle** [comment]
@davidnuzik I see that the UI PUT request to edit the cluster is not sending the new `clusterTemplateRevisionId` - it is still sending the old revision.

**Comment by davidnuzik** [comment]
My bad for not catching that. Thanks @prachidamle 
Updated the issue.

**Comment by davidnuzik** [comment]
Version: master-head (7/25/19) `0c77cb35`

The issue no longer occurs. This appears to have been a caching issue and the PR fixes this.

- I created my template and then created my cluster. UI sends:
clusterTemplateId: "cattle-global-data:ct-x5vjc"
clusterTemplateRevisionId: _"cattle-global-data:validate"_
- I added the second revision to my cluster template, then edited the cluster to switch to it. Now the UI properly sends the new clusterTemplateRevisionId (it did not before):
clusterTemplateId: "cattle-global-data:ct-x5vjc"
clusterTemplateRevisionId: **"cattle-global-data:cloned"**
- Confirmed also do not see rancher server error as well.
- Confirmed cluster properly changes my settings (k8s version in this case) it provisions correctly.
- Smoke test and ensure is functional.
